### ISSUE TITLE
[false-positive] null coalescing

### ISSUE THREAD

**Issue by Koc** [kalessil/phpinspectionsea#1667]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6 |
| Language level | PHP 7.3  |

Current behaviour
--
```php
$fooBar = $this->fooBarService->getFooBar($a->b); // getFooBar returns concrete object of false
$fooBarCode = $fooBar ? $fooBar->code : 'N/A'; // [EA] '$fooBar->code ?? 'N/A'' can be used instead (reduces cognitive load).  suggested here but it is wrong
```

Expected behaviour
--
No false-positive suggestion for above described case.

Environment details
---
PhpStorm 2020.3
Build #PS-203.5981.175, built on December 2, 2020
Licensed to PhpStorm Evaluator
Expiration date: December 27, 2020
Runtime version: 11.0.9+11-b1145.21 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 10.15.7
GC: ParNew, ConcurrentMarkSweep
Memory: 1160M
Cores: 12
Registry: run.processes.with.pty=TRUE, documentation.show.toolbar=true, vcs.annotations.preload=true
Non-Bundled Plugins: CMD Support, FrameSwitcher, Identifier Highlighter Reloaded, Remote call, Scratch, SmarterEditor, String Manipulation, TabSwitch, com.intellij.ideolog, com.jetbrains.annotations.preloader, org.intellij.plugins.postcss, net.seesharpsoft.intellij.plugins.csv, org.github.erikzielke.gotoproject, org.intellij.RegexpTester, name.kropp.intellij.makefile, com.intellij.kubernetes, org.wavescale.sourcesync, socrates.tabshifter, com.jetbrains.lang.ejs, tv.twelvetone.intellij.plugins.intellivue, lermitage.intellij.extra.icons, com.magento.idea.magento2plugin, NEON support, com.dryabov.phpStorm.phpregexp, com.github.woru.options-completion-phpstorm-plugin, com.kalessil.phpStorm.phpInspectionsEA, com.ruslanpolutsygan.adderremover, com.udari, de.espend.idea.php.phpunit, lv.midiana.misc.phpstorm-plugins.deep-keys, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, fr.adrienbrault.idea.symfony2plugin, ru.adelf.idea.dotenv

**Comment by ea-inspections-team** [comment]
Thanks for reporting @Koc, I extracted an example in here: https://3v4l.org/B9SL1 and the proposed replacements working well with false-value. So I'd like to better understand why you consider it to be a false-positive, would you please elaborate more?

**Comment by Koc** [comment]
Indeed. I haven't know that it is legal to combine coalescing operator with property access of booleans.

**Comment by rentalhost** [comment]
I think that it could leads to a false-positive, but maybe rare. For instance: https://3v4l.org/UU5Jo

```php
$test = new stdClass;
$test->code = null;

var_dump($test ? $test->code : 'N/A'); // null
var_dump($test->code ?? 'N/A'); // N/A
```

So I suggest to not use it if the property could be nullable and `null` is in fact a possible result (and here I consider a rare situation).

**Comment by ea-inspections-team** [comment]
@Koc: no worries, it's a rare case.

**Comment by ea-inspections-team** [comment]
@rentalhost: thanks for spotting the breaking behaviour, would you please file a new issue for it with cross-reference to this one? Thanks!
